rust/template/update_handler.rs

```diff
@@ -325,11 +325,11 @@ impl UpdateHandler<Value> for DeltaUpdateHandler
         let handler = self.clone();
         Box::new(move |relid, v, w|{
             let guard_ptr = handler.locked.get();
-            if guard_ptr != ptr::null_mut() {
+            if !guard_ptr.is_null() {
                 let mut guard: Box<MutexGuard<'_, Option<DeltaMap>>> = unsafe {
                     Box::from_raw(guard_ptr as *mut MutexGuard<'_, Option<DeltaMap>>)
                 };
-                (*guard).as_mut().map(|db|db.update(relid, v, w));
+                if let Some(db) = (*guard).as_mut() { db.update(relid, v, w) };
```

Comment:
Do you know why Clippy prefers these?

Reply:
Mostly for readability: https://rust-lang.github.io/rust-clippy/master/index.html#option_map_unit_fn IMO each one looks fine